modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/fields/modal_content/AddFieldsModalContent.tsx

```diff
@@ -65,31 +66,42 @@ const applySavedFDSFields = ({
 	return [selectedKeys, fields];
 };
 
-function filterFields(
-	fields: Array<IFieldTreeItem>,
-	query: string,
-	onFilter?: Function
-) {
+function filterFields({
+	fields,
+	onFilter,
+	onMatch,
+	query,
+}: {
+	fields: Array<IFieldTreeItem>;
+	onFilter?: Function;
+	onMatch?: Function;
```

Comment:
Just recently there was a bug that would have been prevented if I put full args + return value TS instead of `Function`. I told myself I wouldn't be lazy in the future :sweat_smile: Not a strict blocker though, `Function` is still better than nothing.

Reply:
I'm gonna test and push this typing enhancement ❤️ and we are done ✅